packages/core/examples/numericStepperExample.tsx

```diff
@@ -0,0 +1,247 @@
+    private togglePlaceholder = handleBooleanChange((showPlaceholder) => this.setState({ showPlaceholder }));
+    private toggleReadOnly = handleBooleanChange((showReadOnly) => this.setState({ showReadOnly }));
+
+    protected renderOptions() {
+        return [
+            [
+                this.renderMinValueMenu(),
+                this.renderMaxValueMenu(),
+            ], [
+                this.renderButtonPositionMenu(),
+                this.renderIntentMenu(),
+            ], [
+                <label className={Classes.LABEL} key="modifierslabel">Modifiers</label>,
+                this.renderDisabledSwitch(),
+                this.renderReadOnlySwitch(),
+                this.renderLeftIconSwitch(),
+                this.renderPlaceholderSwitch(),
+            ],
+        ];
+    }
+
+    protected renderExample() {
+        return (
+            <div>
+                <NumericStepper
+                    buttonPosition={BUTTON_POSITIONS[this.state.buttonPositionIndex].value}
+                    intent={this.state.intent}
+
+                    min={MIN_VALUES[this.state.minValueIndex].value}
+                    max={MAX_VALUES[this.state.maxValueIndex].value}
+
+                    stepSize={STEP_SIZES[this.state.stepSizeIndex].value}
+                    majorStepSize={MAJOR_STEP_SIZES[this.state.majorStepSizeIndex].value}
+                    minorStepSize={MINOR_STEP_SIZES[this.state.minorStepSizeIndex].value}
+
+                    disabled={this.state.showDisabled}
+                    readOnly={this.state.showReadOnly}
+                    leftIconName={this.state.showLeftIcon ? SAMPLE_LEFT_ICON : null }
+                    placeholder={this.state.showPlaceholder ? SAMPLE_PLACEHOLDER : null }
+
+                    onUpdate={this.handleUpdate}
+                    value={this.state.value}
+                />
+                <br />
+                <div>
+                    <strong>Value:</strong> <code>{this.state.value || "null"}</code>
+                </div>
+            </div>
+        );
+    }
+
+    private renderMinValueMenu() {
+        const { minValueIndex } = this.state;
+        return this.renderSelectMenu("Minimum value", minValueIndex, MIN_VALUES, this.handleMinValueChange);
+    }
+
+    private renderMaxValueMenu() {
+        const { maxValueIndex } = this.state;
+        return this.renderSelectMenu("Maximum value", maxValueIndex, MAX_VALUES, this.handleMaxValueChange);
+    }
+
+    private renderButtonPositionMenu() {
+        const { buttonPositionIndex } = this.state;
+        return this.renderSelectMenu(
+            "Button position", buttonPositionIndex, BUTTON_POSITIONS, this.handleButtonPositionChange);
+    }
+
+    private renderIntentMenu() {
+        return <IntentSelect intent={this.state.intent} key="intent" onChange={this.handleIntentChange} />;
+    }
+
+    private renderDisabledSwitch() {
```

Comment:
Ack. Did it this way because I think it's much easier to grok what's going on when you read the `render` method. Don't feel strongly though.